src/serialization/preprocessor.cpp

```diff
@@ -1156,112 +1162,131 @@ bool preprocessor_data::get_chunk()
-									}
-								}
-							}
-						}
-					}
-
-					if(found_deprecate > 0 && ++found_deprecate == 11) {
-						if(std::equal(buffer.end() - 10, buffer.end(), "deprecated")) {
-							buffer.erase(buffer.end() - 11, buffer.end());
-							skip_spaces();
-							try {
-								DEP_LEVEL level = DEP_LEVEL(std::stoi(read_word()));
-								if(deprecation_level) {
-									deprecation_level = std::max(*deprecation_level, level);
 								} else {
-									deprecation_level = level;
+									parent_.error("Unterminated #arg definition", linenum_);
 								}
-							} catch(const std::invalid_argument&) {
-								// Meh, fall back to default of PREEMPTIVE...
-								deprecation_level = DEP_LEVEL::PREEMPTIVE;
 							}
-							deprecation_version = game_config::wesnoth_version;
-							if(deprecation_level == DEP_LEVEL::PREEMPTIVE || deprecation_level == DEP_LEVEL::FOR_REMOVAL) {
-								skip_spaces();
-								deprecation_version = std::max(deprecation_version, version_info(read_word()));
+						}
+					} else if(directive == "deprecated") {
+						buffer.erase(found_directive - 1, std::string::npos);
+						found_directive = -1;
+						skip_spaces();
+						try {
+							DEP_LEVEL level = DEP_LEVEL(std::stoi(read_word()));
+							if(deprecation_level) {
+								deprecation_level = std::max(*deprecation_level, level);
+							} else {
+								deprecation_level = level;
 							}
+						} catch(const std::invalid_argument&) {
+							// Meh, fall back to default of PREEMPTIVE...
+							deprecation_level = DEP_LEVEL::PREEMPTIVE;
+						}
+						deprecation_version = game_config::wesnoth_version;
+						if(deprecation_level == DEP_LEVEL::PREEMPTIVE || deprecation_level == DEP_LEVEL::FOR_REMOVAL) {
 							skip_spaces();
-							if(!deprecation_detail.empty()){
-								deprecation_detail += '\n';
-							}
-							deprecation_detail += read_rest_of_line();
-							skip_eol();
+							deprecation_version = std::max(deprecation_version, version_info(read_word()));
 						}
-					}
-
-					if(found_enddef > 0 && ++found_enddef == 7) {
-						if(std::equal(buffer.end() - 6, buffer.end(), "enddef")) {
-							break;
-						} else {
-							found_enddef = 0;
-							if(std::equal(buffer.end() - 6, buffer.end(), "define")) { // TODO: Maybe add support for
-																					   // this? This would fill feature
-																					   // request #21343
-								parent_.error(
-										"Preprocessor error: #define is not allowed inside a #define/#enddef pair",
-										linenum);
-							}
+						skip_spaces();
+						if(!deprecation_detail.empty()){
+							deprecation_detail += '\n';
 						}
+						deprecation_detail += read_rest_of_line();
+						skip_eol();
+					} else if(directive == "enddef") {
+						buffer.erase(found_directive - 1, std::string::npos);
+						found_enddef = true;
+						break;
+					} else if(directive == "define") {
+						// TODO: Maybe add support for this? This would fill feature request #21343
```

Comment:
GNA 21343 is Github #1352. However, I think the comment can just be deleted as "wontfix" sounds right to me.